backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -82,6 +82,11 @@ public function putTransaction($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function getCardTransactions($card_id)
+    {
+        return response()->build(self::RESPONSE_MESSAGE_SUCCESS, Transaction::where('card_id', $card_id)->get());
```

Comment:
Instead of returning `Transaction::where('card_id', $card_id)->get()`, we should have a relationship on the Card model to `$card->transactions`. We can then return `$card->transactions` and change the function parameter to `getCardTransactions(Card $card)`.

Reply:
How would you pass a Card into the function though? Is there something in laravel to auto look up the card id in this case?

Reply:
Correct, it will auto select the Card by primary id or throw a 404.

Reply:
I've tried Accepting a Card as a parameter, but I don't think it's looking it up. It just gives me this as a card ```{"imageURL":"http:\/\/via.placeholder.com\/350?text=card+id"}```. There must be something else you have to do for that to work.